### DISCUSSION TITLE
How should use Post Request

### DISCUSSION THREAD

**Discussion by Monirzadeh** [chapar-rest/chapar#58] [Q&A]
i try to use Post Request and get token with this
```
Set Environment Variable
Targey Key: "TokenKey"
Status Code: "200"
From: "From Response"
Path/Key: "message.token"
```
i see token in preview section but when i want use `{{TokenKey}}` in header of of other request it is not working.
did i miss something? i can't find documentation how to use Post Request feature.

**Comment by mirzakhany** [comment]
hi @Monirzadeh 
I think issue is the key that need to start with `$.`  ( dollar and a dot ) to be valid json path ($.message.token). you should be able to see the result in preview section.

I see that is a lake of documentation that I need to fix. 
please let me know if issue still persist after changing the key.

**Reply by Monirzadeh** [reply]
Problem is, when I've set `TokenKey` in Target Key section, I'm not able to use `TokenKey` in other places (e.g. header of other request).

The problem is that I can not use the defined target key anywhere else.
![1](https://github.com/user-attachments/assets/687feff2-52af-42ac-8cef-04bd1b6e7e49)
![2](https://github.com/user-attachments/assets/e3662354-e28a-40f0-a477-4275afc9c805)

**Reply by mirzakhany** [reply]
do you see the TokenKey value populated in your active environment?

**Reply by Monirzadeh** [reply]
No :(
Should convert this discussion to an issue?

**Reply by mirzakhany** [reply]
Yes I think we should 
Please consider this checks before:

- using the latest build
- having the same key as it is in environment 
- Have the target environment selected 
- send the request at least ones and check the active environment 
- 

Another thing you can do is to run project from terminal and see if any specific error is printed out

**Reply by Monirzadeh** [reply]
can you explain this part? i am not sure if i get your point in this one
>Have the target environment selected

other things done

**Reply by mirzakhany** [reply]
On the top right side of the app, there is a drop down to select active environment. That need to be selected when request is sent

**Reply by Monirzadeh** [reply]
so it pass all the checks and issue open in #64

**Comment by mirzakhany** [comment]
I hope this change would fix the issue https://github.com/chapar-rest/chapar/pull/62
